mempool/src/shared_mempool/peer_manager.rs

```diff
@@ -522,41 +522,19 @@ impl PeerManager {
         }
     }
 
-    fn is_primary_upstream_peer(&self, peer: &PeerNetworkId) -> bool {
-        self.upstream_config
-            .get_upstream_preference(peer.raw_network_id())
-            == Some(0)
-    }
-
     fn is_public_downstream(network_id: NetworkId, origin: ConnectionOrigin) -> bool {
         network_id == NetworkId::Public && origin == ConnectionOrigin::Inbound
     }
 
-    // checks whether a peer is a chosen broadcast recipient:
-    // - all primary peers
-    // - fallback peers, if k-policy is enabled
+    // Broadcast to ALL upstream peers, for better reliability of the txn reaching a validator
     // this does NOT check for whether this peer is alive
+    // In the future, we can consider mempool communicating with state sync to detect whether certain
+    // upstream peers are up-to-date enough to broadcast txns to (otherwise the txns might end up not
+    // being ready for broadcast on the upstream peer), but for now we will resort to broadcasting
+    // to all peers across all upstream networks
```

Comment:
might be good to write up an issue with your thoughts on state sync / mempool interrelationship (or a quip)